conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/DoubleArrayList.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib.internal;
+
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import java.util.AbstractList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.RandomAccess;
+
+public final class DoubleArrayList extends AbstractList<Double> implements RandomAccess {
```

Comment:
were using the GNU Trove4J classes considered here?  https://trove4j.sourceforge.net/javadocs/gnu/trove/list/array/TDoubleArrayList.html  Might want to mention in this class's comment why we aren't using those.

Reply:
Hasn't trove been abandonware for the last decade?

Reply:
Ah yes indeed.  Not a decade, but close.  https://bitbucket.org/trove4j/trove/commits/d4433e656b9eba5b1ed713ffe8e15660af37dfb4

Maybe https://eclipse.dev/collections/ then?  I meant this more of a "why build our own?" question than "why not X?"

Reply:
fwiw I think eclipse-collections may be reasonable here as an `implementation` dependency of conjure-java, so long as the eclipse-collection implementation types are never exposed directly. This way we're free to change the underlying implementation as we wish. I believe `new BoxedMutableDoubleList(new DoubleArrayList())` provides the behavior we're looking for.

Reply:
Considered using a OSS implementation of this but some of my first choices were Trove/FastUtils but aren't maintained much. Highbury coincidently ran into a classloader deadlock with eclipse-collections right at the time of writing this 😓. Anyway, due to how small the api needs to be for Conjure access paths, I figured writing it up myself was good.

Conjure gen code only requires an add, addAll, and constructor fwiw. All pretty straightforward to implement, at least as long as Java 21  is concerned.